### PR TITLE
Send user updates to Blink.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request adds an event listener which sends updated users to Blink. It also adds an `iso8601` helper to simplify the kinda frustrating ternaries we were doing before.

#### How should this be reviewed?
References DoSomething/blink#134 for supporting "updated" user events.

#### Checklist
- [ ] Documentation added for changed endpoints.
- [x] Tests added for new features/bug fixes.
- [ ] Post a message in #api if this includes something that causes a rebuild!

### CODE REVIEW

File: app/helpers.php [hunk 1]
```diff
@@ -54,6 +54,19 @@ function format_date($date, $format = 'M j, Y')
...
 
 /**
+ * Format a date as an ISO-8601 timestamp.
+ *
+ * @param Carbon|string $date
+ * @return null|string
+ */
+function iso8601($date)
+{
+    // Fun fact: PHP's built-in DateTime::ISO8601 constant is wrong,
```
Comment (itsjoekent) [inline app/helpers.php]: 💥

File: app/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -34,6 +34,11 @@ public function boot()
         });
 
         User::updating(function (User $user) {
+            // Send payload to Blink for Customer.io profile.
+            if (config('features.blink')) {
+                app(Blink::class)->userCreate($user->toBlinkPayload());
```
Comment (itsjoekent) [inline app/Providers/AppServiceProvider.php]: Im assuming Blink only accepts the `userCreate` and handles it appropriately? just reads weird

Comment (DFurnes) [reply app/Providers/AppServiceProvider.php]: Yeah it totally does! I think it could make sense to rename this endpoint in Blink at some point.

**Comment by DFurnes** [comment]
Well here's a real [head-scratcher](https://app.wercker.com/dosomething/northstar/runs/build/59cd3c818e888f0001ac7564?step=59cd3c87af52880001bc582e):

```
PHPUnit 4.8.36 by Sebastian Bergmann and contributors.

...............................................................  63 / 181 ( 34%)
......E........................................................ 126 / 181 ( 69%)
.......................................................

Time: 1.35 minutes, Memory: 88.00MB

There was 1 error:

1) OAuthTest::testCantRevokeAnotherUsersRefreshToken
ErrorException: Undefined index: access_token

/pipeline/source/tests/Http/OAuthTest.php:511
```

Try as I might, I can't reproduce it on my local. 🤔

**Comment by itsjoekent** [comment]
👀

**Comment by itsjoekent** [comment]
I'll give it a shot, one sec

**Comment by itsjoekent** [comment]
screen shot 2017-09-28 at 2 29 37 pm

Yeah didnt fail for me either

**Comment by DFurnes** [comment]
Ah alright, not sure why it was only happening on Wercker, but the problem was that the Blink & Phoenix mocks were not being recreated when we called `refreshApplication` in that test.

Fixed in 963aaba.

**Comment by DFurnes** [comment]
Rebased to fix a merge conflict, still waiting on corresponding Blink changes to be approved before merging this code in and deploying. 🚀

**Comment by DFurnes** [comment]
Follow-up, I added a feature flag for this in 8894b62 so this is unblocked for merging.

**Comment by itsjoekent** [comment]
new commit seems good to me!

**Pull request merged**